New analysis predicting cell-type-specific effect of Tocilizumab

nrosed opened the pull request:
In this PR I added a new analysis using BuDDI to infer pathway changes after Tocilizumab treatment.

The main files that need to be checked are listed below. The remaining code contains mostly boilerplate code for processing files.

- `sc_preprocessing/0_qc_notebooks/scanpy_analysis_synovium.ipynb` analysis of raw single-cell counts
- `sc_preprocessing/0_qc_notebooks/scanpy_analysis_synovium_bulk.ipynb` analysis of bulk treatment data
- `sc_preprocessing/2_make_pseudobulks/pseudobulk_synovium.ipynb` pseudobulk creation data
- `synovium_analysis/synovium_bulk_hyp_search.ipynb` hyperparameter search
- `synovium_analysis/buddi_synovium_Tocilizumab.ipynb` drug treatment analysis

nrosed committed:
tocilizumab analysis with cell type proportions

nrosed committed:
added tocilizumab analysis

ivichadriana commented on sc_preprocessing/0_qc_notebooks/scanpy_analysis_synovium_bulk.ipynb:
```diff
@@ -0,0 +1,2142 @@
+{
```
<p>Do you want to add a title or description</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1480670129'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

nrosed replied:
<p>added in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/390b86aebf04f040aeb6d49cfc62a7ae77301677" target="_blank">390b86a</a></p>

ivichadriana commented on sc_preprocessing/0_qc_notebooks/scanpy_analysis_synovium_bulk.ipynb:
```diff
@@ -0,0 +1,2142 @@
+{
```
<p>You got two commented out lines here</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1480670130'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='10'/>

nrosed replied:
<p>removed them in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/390b86aebf04f040aeb6d49cfc62a7ae77301677" target="_blank">390b86a</a></p>

ivichadriana commented on sc_preprocessing/0_qc_notebooks/scanpy_analysis_synovium_bulk.ipynb:
```diff
@@ -0,0 +1,2142 @@
+{
```
<p>These and the ones after are commented out</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1480670132'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='16'/>

nrosed replied:
<p>removed in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/390b86aebf04f040aeb6d49cfc62a7ae77301677" target="_blank">390b86a</a></p>

ivichadriana commented on sc_preprocessing/0_qc_notebooks/scanpy_analysis_synovium_bulk.ipynb:
```diff
@@ -0,0 +1,2142 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #2.    del adata.raw
</pre><p>you're deleting the Unnormalized adata?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1480670133'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='26'/>

nrosed replied:
<p>yeah -- i was having issues with it writing to disk and reading it in later</p>

ivichadriana commented on sc_preprocessing/0_qc_notebooks/scanpy_analysis_synovium_bulk.ipynb:
```diff
@@ -0,0 +1,2142 @@
+{
```
<p>Are these commented out on purpose? I think you normalized counts in the other notebook</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1480670135'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='27'/>

nrosed replied:
<p>yes -- because these are bulk data, not single-cell. I do normalization later in the pipeline</p>

ivichadriana commented on sc_preprocessing/2_make_pseudobulks/pseudobulk_synovium.ipynb:
```diff
@@ -0,0 +1,1192 @@
+{
```
<p>Are these commented out on purpose?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1480680824'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='8'/>

nrosed replied:
<p>removed in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/6a0f2690a72842aece33e7952b8fdc09bc9efa06" target="_blank">6a0f269</a></p>

ivichadriana commented on sc_preprocessing/2_make_pseudobulks/pseudobulk_synovium.ipynb:
```diff
@@ -0,0 +1,1192 @@
+{
```
<p>do you want these saved in aug_data_path or cibersort_path</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1480680835'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='13'/>

nrosed replied:
<p>I keep it in aug_data_path, because in my mind, these are "results" needed to understand the augmented data. I could see arguments either way, but this is how I currently do it in my workflow.</p>

ivichadriana commented:
pointed out some minor things you might have missed, as soon as you say it looks OK to you I'll approve! awesome work!!!!!

nrosed committed:
fixed labels and text in notebook

nrosed commented:
<p>updated in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/1ec0950e51f8ed8f83b47cec634cfd58048b9379" target="_blank">1ec0950</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1930948584'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>updated in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/1ec0950e51f8ed8f83b47cec634cfd58048b9379" target="_blank">1ec0950</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1930948592'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>updated in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/1ec0950e51f8ed8f83b47cec634cfd58048b9379" target="_blank">1ec0950</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1930948599'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>yeah - mistake. updated in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/1ec0950e51f8ed8f83b47cec634cfd58048b9379" target="_blank">1ec0950</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1930948612'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>good catch -- i think this is fine though</p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1930948629'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>updated in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/1ec0950e51f8ed8f83b47cec634cfd58048b9379" target="_blank">1ec0950</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1930948640'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>updated in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/1ec0950e51f8ed8f83b47cec634cfd58048b9379" target="_blank">1ec0950</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1930948653'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed committed:
updated text and removed commented out lines

nrosed committed:
removed commented out lines

nrosed commented:
<p>yes, its an explanation why I disable eager execution added text to explain</p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009695'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>no -- at this point gene_df has been filtered such that there is no overlap with cibersort genes, and that their sum is 7000 genes.</p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009780'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>removed commented out line</p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009780'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>ah - I just randomly assing the the cell-type for this plot so that it runs. Ideally, one would predict the highest cell-type and use that. I will update this and fix it in the next release.</p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009796'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed committed:
fixed one plot where font size was too large, removed commented out code, added intro text

nrosed commented:
<p>ack! fixed in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/92d38c3a581f8e31152a5db243e17a8aa194472b" target="_blank">92d38c3</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009847'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>yes -- it is a link explaining why I disabled eager execution</p><p>context added in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/92d38c3a581f8e31152a5db243e17a8aa194472b" target="_blank">92d38c3</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033695'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>context added in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/92d38c3a581f8e31152a5db243e17a8aa194472b" target="_blank">92d38c3</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009695'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>yes, this is correct. removed from file in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/92d38c3a581f8e31152a5db243e17a8aa194472b" target="_blank">92d38c3</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009739'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>moved to top in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/92d38c3a581f8e31152a5db243e17a8aa194472b" target="_blank">92d38c3</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009813'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>fixed in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/92d38c3a581f8e31152a5db243e17a8aa194472b" target="_blank">92d38c3</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009754'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>oops! fixed in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/92d38c3a581f8e31152a5db243e17a8aa194472b" target="_blank">92d38c3</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009677'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>removed all the old variable assignments <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/92d38c3a581f8e31152a5db243e17a8aa194472b" target="_blank">92d38c3</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009716'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p><a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/92d38c3a581f8e31152a5db243e17a8aa194472b" target="_blank">92d38c3</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009780'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>yes -- moved to the top <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/92d38c3a581f8e31152a5db243e17a8aa194472b" target="_blank">92d38c3</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931009831'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed committed:
removed unneeded code and added some more context in headers

nrosed commented:
<p>oops removed in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/7f165f910050fc007c05f91b513e858e5d3054c2" target="_blank">7f165f9</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033708'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>removed in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/7f165f910050fc007c05f91b513e858e5d3054c2" target="_blank">7f165f9</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033728'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>I fixed the code so that I no longer have to write out a giant matrix, so I don't need to comment out those sections anymore, and I deleted them in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/7f165f910050fc007c05f91b513e858e5d3054c2" target="_blank">7f165f9</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033736'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>removed in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/7f165f910050fc007c05f91b513e858e5d3054c2" target="_blank">7f165f9</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033747'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>removed in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/7f165f910050fc007c05f91b513e858e5d3054c2" target="_blank">7f165f9</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033760'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>yeah -- removed since you can see it in the next cell. <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/7f165f910050fc007c05f91b513e858e5d3054c2" target="_blank">7f165f9</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033774'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>I added some context to the comment. I am not sure exactly what it should be, I found a bug doing this previously, were after normalization the correlation between the single-cell and bulk data was very low. I want to keep this in, because I think in the future I will add this to the BuDDI workflow, but I am still unsure what this should look like in practice. comment added in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/7f165f910050fc007c05f91b513e858e5d3054c2" target="_blank">7f165f9</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033788'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>oops updated in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/7f165f910050fc007c05f91b513e858e5d3054c2" target="_blank">7f165f9</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033801'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>I agree --- but it takes too long to remake these plots. If I re-run this analysis, I will fix the plots.</p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033814'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>ah - removed in <a href="https://github.com/greenelab/buddi_analysis/pull/12/commits/7f165f910050fc007c05f91b513e858e5d3054c2" target="_blank">7f165f9</a></p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033820'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

nrosed commented:
<p>yeah -- this was really hard to train. I think its the data that is hard to use because there is a lot of sample-to-sample variability that is difficult to fully capture in the sample latent space. The single-cell reads are also from cite-seq, which actually look quite different than regular single-cell RNA-Seq.  I had to make the pseudobulks in a different way, I made LOTS of pseudobulks using much fewer cells to sample from. This makes noisier pseudobulks, but was the only way to make BuDDI train. I think it's because of the number of reads per cell is very different in CITE_Seq then RNA-Seq, but I didn't do a full comparison. This would be an interesting analysis, because I couldn't find anything online comparing them in depth. (but I only looked for less than an hour)</p>

---

 <a href='https://app.reviewnb.com/greenelab/buddi_analysis/pull/12/discussion/#comment-1931033835'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

ivichadriana reviewed (approved):
all comments addressed, approving
